src/server/server_state.cc

```diff
@@ -70,6 +71,24 @@ void ServerState::Destroy() {
   state_ = nullptr;
 }
 
+bool ServerState::AllowInlineScheduling() const {
+  // We can't allow inline scheduling during a full sync, because then journaling transactions
+  // will be scheduled before RdbLoader::LoadItemsBuffer is finished. We can't use the regular
+  // locking mechanism because RdbLoader is not using transactions.
+
+  // Journal callbacks can preempt; This means we have to disallow inline scheduling
+  // because then we might interleave the callbacks loop from an inlined-scheduled command
+  // and a normally-scheduled command.
+  // The problematic loop is in JournalSlice::AddLogRecord, going over all the callbacks.
+  if (gstate_ == GlobalState::LOADING)
```

Comment:
This check is to make sure we are not the replica? This is a bit confusing with the comment

Reply:
just put the relevant comment above the relevant if and add to the comment if it reference to the replica side or master side